src/zenml/integrations/gcp_secrets_manager/secrets_manager/gcp_secrets_manager.py

```diff
@@ -0,0 +1,260 @@
+    """This function adds the secret group name to the keys of each
+    secret key-value pair to allow using the same key across multiple
+    secrets.
+
+    Args:
+        secret: The ZenML Secret schema
+    """
+    new_dict = dict()
+    for k, v in secret.content.items():
+        new_dict[secret.name + "_" + k] = v
+
+    return new_dict
+
+
+def remove_group_name_from_key(combined_key_name: str, group_name: str) -> str:
+    """This function serves to remove the secret group name from the secret
+    key.
+
+    Args:
+        combined_key_name: Full name as it is within the gcp secrets manager
+        group_name: Group name (the ZenML Secret name)
+    Returns:
+        The cleaned key
+    """
+    if combined_key_name.startswith(group_name + '_'):
+        return combined_key_name[len(group_name + '_'):]
+    else:
+        raise RuntimeError(f"Key-name `{combined_key_name}` does not have the "
+                           f"prefix `{group_name}`. Key could not be "
+                           f"extracted.")
+
+
+@register_stack_component_class
+class GCPSecretsManager(BaseSecretsManager):
+    """Class to interact with the AWS secrets manager."""
+
+    project_id: str
+
+    # Class configuration
+    FLAVOR: ClassVar[str] = "gcp_secrets_manager"
+    CLIENT: ClassVar[Any] = None
+
+    @classmethod
+    def _ensure_client_connected(cls) -> None:
+        if cls.CLIENT is None:
+            cls.CLIENT = secretmanager.SecretManagerServiceClient()
+
+    @property
+    def parent_name(self):
+        return f"projects/{self.project_id}"
+
+    def register_secret(self, secret: BaseSecretSchema) -> None:
+        """Registers a new secret.
+
+        Args:
+            secret: the secret to register"""
+        self._ensure_client_connected()
+
+        if secret.name in self.get_all_secret_keys():
+            raise KeyError(
+                f"A Secret with the name {secret.name} already " f"exists."
+            )
+
+        adjusted_content = add_group_name_to_keys(secret)
+        for k, v in adjusted_content.items():
+            # Create the secret, this only creates an empty secret with the
+            #  supplied name.
+            gcp_secret = self.CLIENT.create_secret(
+                request={
+                    "parent": self.parent_name,
+                    "secret_id": k,
+                    "secret": {
+                        "replication": {"automatic": {}},
+                        "labels": [
+                            (ZENML_GROUP_KEY, secret.name),
+                            (ZENML_SCHEMA_NAME, secret.TYPE),
+                        ],
+                    },
+                }
+            )
+
+            logger.debug("Created empty secret: %s", gcp_secret.name)
+
+            self.CLIENT.add_secret_version(
+                request={
+                    "parent": gcp_secret.name,
+                    "payload": {"data": "{}".format(v).encode()},
```

Comment:
This essentially converts the value into a string right? I think `str(v)` instead of `"{}".format(v)` might be a little clearer at first sight

Reply:
yeah this is a relict of some other str manipulation from before, good catch